core/src/test/java/com/novoda/noplayer/internal/exoplayer/ResumeableAdvertsTest.java

```diff
@@ -0,0 +1,125 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import android.net.Uri;
+
+import com.google.android.exoplayer2.source.ads.AdPlaybackState;
+import com.novoda.noplayer.Advert;
+import com.novoda.noplayer.AdvertBreak;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
+
+import static com.novoda.noplayer.AdvertBreakFixtures.anAdvertBreak;
+import static com.novoda.noplayer.AdvertFixtures.anAdvert;
+import static com.novoda.noplayer.internal.exoplayer.AdGroupFixture.anAdGroup;
+import static org.fest.assertions.api.Assertions.assertThat;
+
+public class ResumeableAdvertsTest {
+
+    private static final int ZERO = 0;
+    private static final int FIVE_SECONDS_IN_MICROS = 5000000;
+    private static final int TEN_SECONDS_IN_MICROS = 10000000;
+    private static final int TWENTY_SECONDS_IN_MICROS = 20000000;
+
+    private static final int HALF_SECOND_IN_MILLIS = 500;
+    private static final int FIVE_SECONDS_IN_MILLIS = 5000;
+    private static final int TEN_SECONDS_IN_MILLIS = 10000;
+    private static final int TWENTY_SECONDS_IN_MILLIS = 20000;
+    private static final int THIRTY_SECONDS_IN_MILLIS = 30000;
+
+    private static final Advert FIRST_ADVERT = anAdvert()
+            .withDurationInMillis(TWENTY_SECONDS_IN_MILLIS)
+            .build();
+    private static final Advert SECOND_ADVERT = anAdvert()
+            .withDurationInMillis(FIVE_SECONDS_IN_MILLIS)
+            .build();
+    private static final Advert THIRD_ADVERT = anAdvert()
+            .withDurationInMillis(TEN_SECONDS_IN_MILLIS)
+            .build();
+
+    private static final AdvertBreak FIRST_ADVERT_BREAK = anAdvertBreak()
+            .withStartTimeInMillis(TEN_SECONDS_IN_MILLIS)
+            .withAdverts(FIRST_ADVERT, SECOND_ADVERT)
+            .build();
+    private static final AdvertBreak SECOND_ADVERT_BREAK = anAdvertBreak()
+            .withStartTimeInMillis(TWENTY_SECONDS_IN_MILLIS)
+            .withAdverts(THIRD_ADVERT)
+            .build();
+    private static final List<AdvertBreak> ADVERT_BREAKS = Arrays.asList(FIRST_ADVERT_BREAK, SECOND_ADVERT_BREAK);
+    private static final AdPlaybackState.AdGroup SECOND_AD_GROUP = secondAdGroupFixture().build();
+
+    @Test
+    public void createsAdvertPlaybackStateWithResumePositionInMicroseconds() {
```

Comment:
This is the test case that breaks. If we have the first advert having a duration greater than the resume position and then the second advert having a duration that is less than the resume position.